doc/app_devel_docs.rst

```diff
@@ -0,0 +1,337 @@
+
+      (_,e) <- el' "div" $ text "Hello"
+      
+      let getCoords e = DOM.liftJSM $ do
+            rect <- DOM.getBoundingClientRect (_raw_element e)
+            y <- DOM.getY rect
+            h <- DOM.getHeight rect
+            return (y,h)
+
+      performEvent (getCoords e <$ ev)
+
+* But when using external .js files, one has to do arbitrary JS code execution.
+
+  For doing this ``jsaddle`` package is preferred as it provides a type-safe way to execute the JS code.
+
+  See `documentation <https://hackage.haskell.org/package/jsaddle-0.9.4.0/docs/Language-Javascript-JSaddle-Object.html>`_ of ``Language.Javascript.JSaddle.Object`` for examples
+
+  See :ref:`dom_ui_libs` for example usage.
+
+* It is also possible to do arbitrary JS code block execution using ``eval`` API from ``Language.Javascript.JSaddle.Evaluate``. ::
+
+    eval :: (ToJSString script) => script -> JSM JSVal
+
+    liftJSM $ eval "console.log('Hello World')"
+
+* JSFFI functions
+
+  This will only work with ``ghcjs``::
+
+    import GHCJS.Types (JSVal)
+    
+    foreign import javascript unsafe
+      "try { $r = $1 / $2; } catch (e) { $r = "error"; }"
+      divide :: Double -> Double -> JSVal
+
+  See https://github.com/ghcjs/ghcjs/blob/master/doc/foreign-function-interface.md
+
+If you access a DOM element via FFI which has just been created, then it might result in an error. This is because the ``getPostBuild`` is fired before the DOM has been put in the Window. To solve this add ``delay``::
+
+  evPB <- delay 0.2 =<< getPostBuild
+  elAttr ("id" =: "some-uniq-id") $ text "example"
+  performEvent_ (someFFI <$ evPB)
+
+Capturing DOM events with FFI
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Many of the Web APIs work on a `callback` mechanism, where a user supplied function will be called. Many of these APIs in JS code start with `on` prefix.
+
+Example JS code for creating an AudioNode to handle audio data,
+`Source <https://developer.mozilla.org/en-US/docs/Web/API/ScriptProcessorNode>`_ ::
+
+  // Give the node a function to process audio events
+  scriptNode.onaudioprocess = function(audioProcessingEvent) {
+    // The input buffer is the song we loaded earlier
+    var inputBuffer = audioProcessingEvent.inputBuffer;
+    ..
+    }
+
+Similar callback can be created by using the ``on`` API from ``GHCJS.DOM.EventM`` ::
+
+  -- here audioProcess is the equivalent "tag" for JS onaudioprocess
+
+    myNode :: ScriptProcessorNode
+    
+    liftJSM $ on myNode audioProcess myAudioProcessHandler
+
+  myAudioProcessHandler :: EventM ScriptProcessorNode AudioProcessingEvent ()
+  myAudioProcessHandler = do
+    -- aEv :: AudioProcessingEvent
+    aEv <- ask
+    buf <- getInputBuffer aEv
+    ..
+
+Exception Handling
+~~~~~~~~~~~~~~~~~~
+
+..   Add proper exception handling
+
+     jsFn <- eval "(function (cb) { try{cb();alert(1)}catch(e){console.warn(e);alert(2)} })"
+     
+         (funJsv) <- function $ \ _ _ args -> do
+           io $ print 13333
```

Comment:
Is it just me or is the indentation here a little off?

Reply:
This is actually commented. I don't have any good stuff to add in this section. So this is blank section.